host/sgx/create.c

```diff
@@ -1075,14 +1094,21 @@ oe_result_t oe_create_enclave(
     OE_CHECK(oe_sgx_initialize_load_context(
         &context, OE_SGX_LOAD_TYPE_CREATE, flags));
 
-#ifdef OE_WITH_EXPERIMENTAL_EEID
     for (size_t i = 0; i < setting_count; i++)
+    {
+        if (settings[i].setting_type == OE_SGX_ENCLAVE_CONFIG_DATA)
+        {
+            context.config_data = (oe_config_data_t*)settings[i].u.config_data;
```

Comment:
I am doubtful here as I am typecasting const to a non-const pointer. Can you suggest better alternatives.

Reply:
I think you have a good intuition that casting to remove `const` is generally a bad practice; in this case, your code change defines both the new settings data structure (as const) and the context data structure member (as non-const), so I think the better question is why define those differently. It looks like context does not expect to modify the config data either, so it could be `const` as well.

As an extra consideration for more general cases it usually boils down to a couple of cases:
- The original intent was always that the parameter or member needs to be modified, in which case the source definition should just be non-const.
- The target does not need to be modified and so also can be marked const (this case)
- The source is intended to be const, but we need to modify the information for some reason. In this case, you should usually pass by value instead of reference (i.e. make a non-const copy of the data).